emails/views.py

```diff
@@ -268,6 +269,14 @@ def _sns_message(message_json):
 
     try:
         relay_address = RelayAddress.objects.get(address=local_portion)
+
+        '''
+        reply_address = ReplyAddress.objects.get_or_create(
+            relay_address=relay_address, sender=sha256(from_address).hexdigest()
+        )
+        replace Reply-To: with reply_address.address
```

Comment:
We also need to replace `Return-Path` header.

Reply:
We should also look at the `Received` header value, which could leak the mail *provider* of the person sending the reply.

Reply:
`Authentication-Results`, `Received-SPF`, `DKIM-Signature` could also leak the *provider* of the person sending a reply.